src/main/java/com/cloudbees/jenkins/support/SupportPlugin.java

```diff
@@ -171,7 +174,26 @@ public class SupportPlugin extends Plugin {
     public SupportPlugin() {
         super();
         handler.setLevel(getLogLevel());
-        handler.setDirectory(getRootDirectory(), "all");
+        handler.setDirectory(getLogsDirectory(), "all");
+    }
+
+    @Initializer(after = InitMilestone.EXTENSIONS_AUGMENTED)
+    public static void migrateExistingLogs() {
+        File rootDirectory = getRootDirectory();
+        File[] files = rootDirectory.listFiles();
+        if (rootDirectory != null) {
```

Comment:
@Vlatombe I think you meant:

```suggestion
        if (files != null) {
```

All test are currently failing with:

```
java.lang.NullPointerException: Cannot read the array length because "<local3>" is null
	at com.cloudbees.jenkins.support.SupportPlugin.migrateExistingLogs(SupportPlugin.java:185)
Caused: java.lang.reflect.InvocationTargetException
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:568)
	at hudson.init.TaskMethodFinder.invoke(TaskMethodFinder.java:109)
Caused: java.lang.Error
	at hudson.init.TaskMethodFinder.invoke(TaskMethodFinder.java:115)
	at hudson.init.TaskMethodFinder$TaskImpl.run(TaskMethodFinder.java:185)
	at org.jvnet.hudson.reactor.Reactor.runTask(Reactor.java:305)
	at jenkins.model.Jenkins$5.runTask(Jenkins.java:1161)
	at org.jvnet.hudson.reactor.Reactor$2.run(Reactor.java:222)
	at org.jvnet.hudson.reactor.Reactor$Node.run(Reactor.java:121)
	at jenkins.security.ImpersonatingExecutorService$1.run(ImpersonatingExecutorService.java:70)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)
	at java.base/java.lang.Thread.run(Thread.java:833)
```